devicetypes/stelpro/stelpro-maestro-thermostat.src/stelpro-maestro-thermostat.groovy

```diff
@@ -0,0 +1,617 @@
+			state "configure", label:'', action:"configuration.configure", icon:"st.secondary.configure"
+		}
+
+		main ("thermostatMulti")
+		details(["thermostatMulti", "humidity", "heatingSetpoint", "refresh", "configure"])
+	}
+}
+
+def parse(String description) {
+	return parseCalled(description)
+}
+
+def parseCalled(String description) {
+	log.debug "Parse description $description"
+	def map = [:]
+	if (description?.startsWith("read attr -")) {
+		def descMap = parseDescriptionAsMap(description)
+		log.debug "Desc Map: $descMap"
+		if (descMap.cluster == "0201" && descMap.attrId == "0000")
+		{
+			map.name = "temperature"
+			map.value = getTemperature(descMap.value)
+			if (descMap.value == "7ffd") {		//0x7FFD
+				map.value = "low"
+			}
+			else if (descMap.value == "7fff") {	//0x7FFF
+				map.value = "high"
+			}
+			else if (descMap.value == "8000") {	//0x8000
+				map.value = "--"
+			}
+			
+			else if (descMap.value > "8000") {
+				map.value = -(Math.round(2*(655.36 - map.value))/2)
+			}
+						
+			sendEvent(name:"temperature", value:map.value)
+		}
+		else if (descMap.cluster == "0201" && descMap.attrId == "0012") {
+			log.debug "HEATING SETPOINT"
+			map.name = "heatingSetpoint"
+			map.value = getTemperature(descMap.value)
+			if (descMap.value == "8000") {		//0x8000
+				map.value = "--"
+			}
+			sendEvent(name:"heatingSetpoint", value:map.value)
+		}
+	   /* else if (descMap.cluster == "0201" && descMap.attrId == "001c") {
+			if (descMap.value.size() == 8) {
+				log.debug "MODE"
+				map.name = "thermostatMode"
+				map.value = getModeMap()[descMap.value]
+				sendEvent(name:"thermostatMode", value:map.value)
+			}
+			else if (descMap.value.size() == 10) {
+				log.debug "MODE & SETPOINT MODE"
+				def twoModesAttributes = descMap.value[0..-9]
+				map.name = "thermostatMode"
+				map.value = getModeMap()[twoModesAttributes]
+				sendEvent(name:"thermostatMode", value:map.value)
+			}
+		}
+		else if (descMap.cluster == "0201" && descMap.attrId == "401c") {
+			log.debug "SETPOINT MODE"
+			log.debug "descMap.value $descMap.value"
+			map.name = "thermostatMode"
+			map.value = getModeMap()[descMap.value]
+			sendEvent(name:"thermostatMode", value:map.value)
+		}*/
+		else if (descMap.cluster == "0201" && descMap.attrId == "0008") {
+			log.debug "HEAT DEMAND"
+			map.name = "thermostatOperatingState"
+			if (descMap.value < "10") {
+				map.value = "idle"
+			}
+			else {
+				map.value = "heating"
+			}
+			sendEvent(name:"thermostatOperatingState", value:map.value)
+			if (settings.heatdetails == "No") {
+				map.displayed = false
+			}
+		}
+	}
+	else if(description?.startsWith("humidity")) {
+		log.debug "DEVICE HUMIDITY"
+		map.name = "humidity"
+		map.value = (description - "humidity: " - "%").trim()
+		sendEvent(name:"humidity", value:map.value)
+	}
+
+	def result = null
+	if (map) {
+		result = createEvent(map)
+	}
+	log.debug "Parse returned $map"
+	return result
+}
+
+def parseDescriptionAsMap(description) {
```

Comment:
Should be able to use `zigbee. parseDescriptionAsMap` instead of redefining

Reply:
As long as it works, I'm fine with that